### ISSUE TITLE
MDCCollectionViewStyler cellBackgroundColorAtIndexPath doesn't support dark mode

### ISSUE THREAD

**Issue by rkgibson2** [material-components/material-components-ios#9522]
If the color returned from `-cellBackgroundColorAtIndexPath:` is an iOS 13 dynamic color, the resulting cell background will not change color when the mode is switched.

## Reproduction steps

1. Return an iOS 13 dynamic color from the `-cellBackgroundColorAtIndexPath:` delegate method.
2. Load the cell.
3. Change the device color appearance from light to dark (or vice versa).

## Expected behavior

The cell should now have a background color suitable for the new color appearance.

## Actual behavior

The cell will have the same background color as in the initial appearance.

## Platform

 - Device: All
 - OS: iOS 13

## Additional context

Looking at the code, it looks like the issue may be that the background color is displayed as a UIImage of the given color, and that image is cached. If the image is generated in light mode, it will use the light mode color. When the color mode is changed, the cache isn't cleared, so the old image will be used again, with the wrong color.

**Comment by randallli** [comment]
Hi rkgibson2, is this blocking your release?

**Comment by rkgibson2** [comment]
No, it's not blocking anything. We have a workaround (resolve the colors before returning from `-cellBackgroundColorAtIndexPath:` so the cache works correctly), but it'd be nice to remove that eventually.
